tdrs-backend/tdpservice/users/serializers.py

```diff
@@ -48,17 +54,29 @@ class Meta:
         extra_kwargs = {"password": {"write_only": True}}
 
 
-class SetUserProfileSerializer(serializers.ModelSerializer):
+class UserProfileSerializer(serializers.ModelSerializer):
     """Serializer used for setting a user's profile."""
 
+    stt = STTUpdateSerializer(required=True)
+    email = serializers.SerializerMethodField("get_email")
+
     class Meta:
         """Metadata."""
 
         model = User
-        fields = ["first_name", "last_name"]
+        fields = ["first_name", "last_name", "stt", "email"]
```

Comment:
In order to add STT to auth-check, and email to set-profile, so they'd have the same payload.